stablehlo/dialect/StablehloOps.cpp

```diff
@@ -2861,6 +2861,98 @@ LogicalResult ClampOp::reifyReturnTypeShapes(
 // ComplexOp
 //===----------------------------------------------------------------------===//
 
+namespace {
+// Utility function, used by printSelectOpType and
+// printSameOperandsAndResultType. Given a FunctionType, assign the types
+// to operands and results, erroring if any mismatch in number of operands
+// or results occurs.
+ParseResult assignFromFunctionType(OpAsmParser& parser, llvm::SMLoc loc,
+                                   ArrayRef<Type*> operands, Type& result,
+                                   FunctionType& fnType) {
+  assert(fnType);
+  if (fnType.getInputs().size() != operands.size()) {
+    return parser.emitError(loc)
+           << operands.size() << " operands present, but expected "
+           << fnType.getInputs().size();
+  }
+
+  // Set operand types to function input types
+  for (auto [operand, input] : llvm::zip(operands, fnType.getInputs())) {
+    *operand = input;
+  }
+
+  // Set result type
+  if (fnType.getResults().size() != 1) {
+    return parser.emitError(loc, "expected single output");
+  }
+  result = fnType.getResults()[0];
+
+  return success();
+}
+
+// getInferredComplexType takes a complex tensor type and returns a
+// type that maintains the shape, but removes the complex type for the
+// underlying data type
+//   Ex: tensor<4xcomplex<f32>>  -->  tensor<4xf32>
+Type getInferredComplexType(Type result) {
+  assert(result.isa<TensorType>() &&
+         result.cast<TensorType>().getElementType().isa<ComplexType>());
+  TensorType tensorTy = result.cast<TensorType>();
+  ComplexType complexTy = tensorTy.getElementType().cast<ComplexType>();
+  Type elementTy = complexTy.getElementType();
+
+  return hlo::getSameShapeTensorType(tensorTy, elementTy);
+}
+
+// ComplexOpType - only print result type if the inferred complex type
+// matches all operand types.
+//
+// Inferring operand types for complex ops:
+//  %0 = mhlo.complex %1, %2 : tensor<4xcomplex<f32>>
+//    %0 : tensor<4xcomplex<f32>>
+//    %1 : tensor<4xf32>
+//    %2 : tensor<4xf32>
+void printComplexOpType(OpAsmPrinter& p, Operation* op, Type lhs, Type rhs,
+                        Type result) {
+  Type inferredResult = getInferredComplexType(result);
```

Comment:
This doesn't look like a result, since we're matching it to with lhs and rhs types. Consider renaming to `realType` or `inferredOperandType`. (Same for `Type inferredTy` below).

Reply:
True. Going with `realType` since it is accurate and precise.